src/transactions/feed/queryHelper.ts

```diff
@@ -180,11 +225,19 @@ export function useFetchTransactions(): QueryHookResult {
     // 2. sometimes blockchain-api returns 0 transactions for a page (as we only
     //    display certain transaction types in the app) and for this case,
     //    automatically fetch the next page(s) until some transactions are returned
+    //
+    // This has the effect of setting the fetchingMoreTransactions flag to true iff
+    // - We are not already loading
+    // - There exists at least one chain that has futher pages
+    // - EITHER we do not yet have enough TXs, OR NO chains whatsoever produced results
+    //    in the most recent round of fetching (which corresponds to case 2. above
+    //    occuring for all chains simaltaneously)
     const { transactions, pageInfo, hasTransactionsOnCurrentPage } = fetchedResult
     if (
       !loading &&
-      pageInfo?.hasNextPage &&
-      (transactions.length < MIN_NUM_TRANSACTIONS || !hasTransactionsOnCurrentPage)
```

Comment:
I don't get why this is an OR, and it makes me uneasy to see the changes not understanding why it was done this way to begin with. Speaking of the original code only-- why would you fetch another page if you had already reached the minimum number of transactions to display? And now of the new changes-- are you sure that more transactions should be fetched only if *all* chains have no transactions on the current page?

Reply:
My understanding was that the original intent of this hook is to cover two cases:

1. We're fetching initial transactions to populate the home feed, and we don't have enough yet -- keep fetching more pages until we have enough. 
2. We're fetching transactions _for any reason_ (whether through polling or an explicit request) and the current page that we got back from blockchain-api contains _no_ transactions at all, for whatever reason. This is indeed possible. Fetch the next page to "make up" for this empty page. For example, if a user swipes down to load more TXs and there are in fact more TXs available, but blockchain-api returns an intermediate empty page, the user shouldn't have to swipe again.

In both these cases, subsequent requests will only be made if more pages actually exist. In the move to multichain, nothing needs to really change for case 1. For case 2, what we're trying to avoid is _no_ transactions appearing despite more pages being available; for multichain, this occurs when no chains produce TXs on their most-recently-fetched page, but at least one chain has further pages remaining.